plugins/catalog-react/src/routes.ts

```diff
@@ -37,16 +42,54 @@ export const entityRouteRef = getOrCreateGlobalSingleton(
     }),
 );
 
+/**
+ * Configurable options for `entityRouteParams`
+ * @public
+ */
+export type EntityRouteParamsOptions = {
+  encodeParams: boolean;
+};
+
 /**
  * Utility function to get suitable route params for entityRoute, given an
  * @public
  */
-export function entityRouteParams(entity: Entity) {
+export function entityRouteParams(
+  entityOrRef: Entity | CompoundEntityRef | string,
+  options: EntityRouteParamsOptions = { encodeParams: false },
```

Comment:
make the assignment (defaults) inside the function body instead - leaving it visible in the contract actually has surprising effects

Reply:
Removed from the contract.

Additionally, If you have a link or a doc on those surprising effects, I'd love to learn more.

Reply:
You know what, thanks for pushing on that because this led me down a deep dark rabbit hole of the js language specification :) And I was wrong! Indeed they've pinned down that those arguments are evaluated at call time, _every time_, including complex defaults. I was so hurt by python subtly stabbing my back with the way that its defaults are evaluated at _definition time_ leading to subtle bugs unless you're careful. And I transferred that onto js. But for all its flaws, that was not one of them.